docs: Identity Center Okta to Teleport migration guide

tcsc opened the pull request:
- Adds a migration guide to integrate with an existing Okta-managed Identity Center configuration
- Organizes identity center related docs under `/aws-iam-identity-center/` path. 
- Removes the "preview" label from the guide.

Before:
```
/guides
/guides/aws-iam-identity-center.mdx
/guides/datadog.mdx
/guides/*
```

Now:
```
/guides
/guides/aws-iam-identity-center/aws-iam-identity-center.mdx
/guides/aws-iam-identity-center/guide.mdx
/guides/aws-iam-identity-center/migrating-identity-center-from-okta-to-teleport.mdx
/guides/datadog.mdx
/guides/*

```

I checked the references to identity center docs so far and all looks good, i.e. no need for forced redirects.

tcsc committed:
docs: Identity Center Okta to Teleport migration guide

r0mant commented on docs/img/identity-center/ic-migrate-end.png:
Let's please use LucidChart for the diagrams so all the graphics in our documentation are consistent.

r0mant commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,211 @@
+Identity Center users and groups to Teleport management, including having
+Teleport control individual user and group Account Assignments.
+
+By default, Teleport wants to control all user and group provisioning into 
```
```suggestion
By default, Teleport controls all user and group provisioning into 
```

r0mant commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,211 @@
+ - watching the Teleport logs
+
+Once your Okta users are imported into Teleport, you can progress to the next
+step:
```
```suggestion
step.
```

r0mant commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,211 @@
+sync has occurred. You can verify this by either
+ - refreshing the user page and finding your Okta users,
+ - checking the Okta integration status page, or
+ - watching the Teleport logs
```
Nit: I'd remove this one in favor of the previous two more user-friendly options. Plus, it won't suit Cloud customers.
```suggestion
```

r0mant commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,211 @@
+    --url ${IDENTITY_CENTER_SCIM_BASE_URL} \
+    --token ${IDENTITY_CENTER_SCIM_BEARER_TOKEN} \
+    --default-owner ${TELEPORT_ACCESS_LIST_DEFAULT_OWNER} \
+    --user-origin okta \
+    --account-name ${ACCOUNT_NAME_ALLOW_FILTER} \
+    --group-name ${GROUP_NAME_ALLOW_FILTER}
```
A logical question that arises, how can these filters be updated after the integration has been created? I didn't see a mention of this later in the guide.

r0mant commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,211 @@
+
+For more information, see the [Identity Center Integration guide](./aws-iam-identity-center.mdx)
+
+## Step 6: Retire Okta group provisioning
```
Should probably also include a section on how to unenroll the Teleport integration side of this as well in case someone decides they don't want to switch yet.

r0mant reviewed (commented):
Looks reasonable, left a few things to consider.

tcsc committed:
Apply suggestions from code review

Co-authored-by: Roman Tkachenko <roman@goteleport.com>

tcsc committed:
Updated diagrams

tcsc committed:
Merge remote-tracking branch 'origin/master' into tcsc/ic-migration-guide

tcsc committed:
Review feedback

- Rearranged to move migration path into "how it works"
- Added Uninstall section
- Added edit section

tcsc committed:
Merge branch 'master' into tcsc/ic-migration-guide

tcsc committed:
cspell update

zmb3 commented on docs/img/identity-center/ic-migrate-mid.png:
Why does this one have a white background and the other images have transparent backgrounds?

tcsc replied:
No idea. Will re-export from lucid and see what happens.

zmb3 commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
````diff
@@ -0,0 +1,325 @@
+use `tctl` to install it:
+
+```console
+$ ./tctl plugins install okta \
````
```suggestion
$ tctl plugins install okta \
```

zmb3 commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,325 @@
+To make sure everything is working, wait until the first Okta to Teleport user
+sync has occurred. You can verify this by either
+ - refreshing the user page and finding your Okta users,
+ - checking the Okta integration status page, or
```
or what?

zmb3 commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
````diff
@@ -0,0 +1,325 @@
+Again, we need to install the plugin using `tctl`. 
+
+```console
+$ ./tctl plugins install awsic \
````
```suggestion
$ tctl plugins install awsic \
```

zmb3 commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,325 @@
+
+### Group import control
+
+The Group import allow-list is controlled by the `--group-name` option. You can
```
```suggestion
The Group import allow list is controlled by the `--group-name` option. You can
```

zmb3 commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,325 @@
+groups you want Okta to maintain control over.
+</Admonition>
+
+<Admonition type="warning">
```
Stacking admonitions like this creates a lot of noise. I would prefer more plain text and only use the admonition when there's something you really need to highlight.

zmb3 commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,325 @@
+   - View applications and their details
+ - The AWS Identity Center ARN, AWS region, SCIM base address and SCIM bearer token
+
+## Step 1: Install Okta SAML connector
```
Try to use the `Step 1/N` format for consistency with other guides.

r0mant commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,325 @@
+
+We will be using a very limited subset of the Teleport Okta integration in this 
+deployment, disabling all features except periodic user synchronization. This
+configuration is not supported by the normal installation UI, so we will have to
```
```suggestion
configuration is not currently supported by the normal installation UI, so we will have to
```

r0mant commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
````diff
@@ -0,0 +1,325 @@
+the integration's plugin resource with `tctl`.
+
+```console
+./tctl edit plugins/aws-identity-center
````
```suggestion
$ tctl edit plugins/aws-identity-center
```

r0mant commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
````diff
@@ -0,0 +1,325 @@
+      region: us-east-1
+```
+
+You can add or remove filters to the various filters. Once you save and quit the
````
"filters to the various filters" reads strange

tcsc replied:
And this is why proofreading exists.

r0mant commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,325 @@
+
+You can add or remove filters to the various filters. Once you save and quit the
+editor, `tctl` will replace the existing resource with your updated version. This
+will automatically restart the Identity Center Integration with the ne2 filters.
```
```suggestion
will automatically restart the Identity Center Integration with the new filters.
```

tcsc committed:
Apply suggestions from code review

Co-authored-by: Roman Tkachenko <roman@goteleport.com>
Co-authored-by: Zac Bergquist <zac.bergquist@goteleport.com>

tcsc committed:
Applied feedback

tcsc committed:
Apply feedback

tcsc committed:
Apply feedback

smallinsky commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,334 @@
+$ tctl plugins install awsic \
+    --arn ${IDENTITY_CENTER_INSTANCE_ARN} \
+    --region ${IDENTITY_CENTER_INSTANCE_REGION} \
+    --use-system-credentials \
```
Can we add more details on use-system-credent? 
Like provide hint that it refers to AWS credentials. Specifically, it should clarify that use-system-credent will allow to load AWS credentials from the local environment in Teleport Auth to authenticate with the AWS IC API.

r0mant commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
````diff
@@ -0,0 +1,334 @@
+will have to use `tctl` to install it:
+
+```console
+$ tctl plugins install okta \
````
Is there a command reference for this command and "install awsic" used below? I think we should have it to explain all the flags: https://goteleport.com/docs/reference/cli/tctl/.

Here (and below where "install awsic" is used), it might be helpful to briefly explain what effect these flags have as well.

tcsc replied:
Added in #52878

r0mant commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,334 @@
+Deleting the integration will automatically remove all of the Teleport resources 
+used by the integration to manage its own state, as well as all of the Teleport
+roles created by the integration used to grant AWS Identity Center account
+assignments and Access Lists imported from AWS Identity Center Groups.
```
Is this true? Do we actually remove all imported access lists as well? I think we keep them for Okta integration, right? Just making sure this is accurate (and does the UI warn user about what will be removed? it definitely should cc @flyinghermit)

Also, what happens with access lists with IC-assigned-roles you created yourself that get pushed to IC? Those aren't removed, correct? This should be clarified here as well.

tcsc replied:
Can confirm that this is the current behaviour. No warnings in `tctl plugins delete` at the moment.

tcsc replied:
User created roles and access lists will not be deleted. If a user created access list refers to a plugin-created role, it will be left with a dangling role reference once the integration is deleted, and will need manual cleanup.

I have a bunch of `// TODO:`s in various places for walking User and Access List role assignments and clean them up when deleting plugin-created roles, but nothing actually implemented for it. 

Will add some notes about the above to the doc.

r0mant replied:
Can you create a ticket to add confirmation/warning to tctl plugins delete? Maybe @flyinghermit can address it at the same time we add a similar warning to the UI.

flyinghermit replied:
Ticket added https://github.com/gravitational/teleport.e/issues/6165

r0mant reviewed (approved):
lgtm with a couple remaining comments

tcsc committed:
Merge branch 'master' into tcsc/ic-migration-guide

tcsc committed:
Adds decription of installation commands

tcsc committed:
Merge remote-tracking branch 'origin/master' into tcsc/ic-migration-guide

tcsc committed:
Merge remote-tracking branch 'origin/tcsc/ic-migration-guide' into tcsc/ic-migration-guide

tcsc committed:
linter appeasement

tcsc committed:
Merge remote-tracking branch 'origin/master' into tcsc/ic-migration-guide

tcsc committed:
Merge branch 'master' into tcsc/ic-migration-guide

tcsc committed:
Merge branch 'master' into tcsc/ic-migration-guide

mmcallister commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,359 @@
+tocDepth: 3
+---
+
+The goal of this guide is to allow you to perform a staged migration from
```
Moving to active voice to keep consistent:

This guide helps you migrate AWS Identity Center users and groups from Okta to Teleport in stages while keeping Okta as your AWS SSO provider

mmcallister commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,359 @@
+Okta-managed AWS Identity Center users & groups, to having them managed by Teleport
+while preserving Okta as your AWS SSO provider.
+
+This approach lets you get set up with a minimal Teleport AWS Identity Center
```
This approach sets up a minimal Teleport AWS Identity Center integration that you can expand as you become more familiar with Teleport. It also provides deployment options unavailable in a standard Teleport Identity Center setup, such as deploying Teleport in a private network.

mmcallister commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,359 @@
+"standard" Teleport Identity Center configuration, such as having Teleport deployed
+in a private network.
+
+You can also use this setup if you don't want to migrate fully to using Teleport
```
Use this setup if you prefer not to fully migrate to Teleport as an Identity Provider for AWS Identity Center. By following this guide, you can let Teleport manage user and group AWS account assignments while keeping Okta as AWS Identity Center’s IdP.

mmcallister commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,359 @@
+
+## Deleting the Identity Center integration
+
+If you decide not to switch over to Teleport you can delete the Identity Center
```
If you decide not to switch to Teleport, you can delete the Identity Center integration in two ways.

Deleting the integration automatically removes all Teleport resources it used to manage its state, including:

- Teleport roles created for AWS Identity Center account assignments
- Access Lists imported from AWS Identity Center groups

However, user-created resources remain intact, including:

- Access Lists you created
- Roles you manually configured for account assignments

If an Access List grants permissions to a now-deleted integration role, or if a user has a deleted role assigned directly, you must manually remove those references.

mmcallister commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
````diff
@@ -0,0 +1,359 @@
+    --no-appgroup-sync
+```
+
+This will install the Okta integration and start the User sync service with a
````
This will install the Okta integration and start the user sync service with a
configuration that:
- Imports Okta users assigned to the Okta App `${OKTA_SAML_APP_ID}`, and keeps them synced with the upstream Okta organization.
- Does *not* expose a SCIM service.
- Does *not* attempt to sync or manage any other resources from Okta.

mmcallister commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,359 @@
+ - Okta and Teleport both control user provisioning via SCIM
+ - Teleport manages a second subset of Identity Center groups (selected via group filters during plugin installation)
+ - Teleport controls IC Group Account Assignments for the Identity Center groups under its control
+ - Teleport controls direct IC User Account Assignments for the Identity Center users under its control
```
Teleport controls direct IC user account Assignments for the Identity Center users under its control
Comment

(lowercase, following our previous standard in other docs, like in /okta/hosted-guide)

mmcallister commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,359 @@
+assignments.
+
+<Admonition type="warning">
+Individual User Account Assignments will _not_ be preserved during import. You
```
Individual user account Assignments will _not_ be preserved during import.

mmcallister commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,359 @@
+Example filters:
+ - `administrators`: The literal "administrators" group
+ - `site-*`: Any group with the prefix `site-`
+ - `^(?:[^a]|a[^w]|aw[^s]|aws[^\-]).*$`: Any group that does _not_ have the prefix `aws-`
```
***not***

mmcallister commented:
Feedback provided, minor edits

flyinghermit commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,359 @@
+This is fine for most deployments, but can cause difficulties when deploying into
+a sophisticated, Okta-managed AWS Identity Center configuration. 
+
+By contrast, the deployment in this guide will transition transition control
```
```suggestion
By contrast, the deployment in this guide will transition control
```

flyinghermit commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,359 @@
+a sophisticated, Okta-managed AWS Identity Center configuration. 
+
+By contrast, the deployment in this guide will transition transition control
+from Okta to Teleport in a staged fashion.
```
nit: elaborate on what is expected
```suggestion
from Okta to Teleport in a staged fashion where user identities are managed by Okta but account permission assignment is managed by Teleport.
```

flyinghermit commented on docs/pages/admin-guides/management/guides/migrating-iam-ic-from-okta-to-teleport.mdx:
```diff
@@ -0,0 +1,359 @@
+## Prerequisites
+
+ - A working Teleport cluster
+ - An AWS role configured as per the [Identity Center guide](./aws-iam-identity-center.mdx)
```
I think we should scope down the sub-heading link so reader can dive straight into it instead of reading all of that doc. E.g. `(./awsxxxx.mdx#sub-heading)`

flyinghermit committed:
address review comments

flyinghermit committed:
organize ic docs

flyinghermit committed:
move how to guide to root docs page, add detailed deletion guide to getting started page

flyinghermit commented:
@mmcallister I addressed all of your suggestions in https://github.com/gravitational/teleport/pull/51861/commits/a533e461ee6bfcc07acaece8c04cf91a0442fa66. I also updated the structure of the docs so that the identity center related docs are grouped under `/aws-iam-identity-center/` path. Please take a look again.

mmcallister reviewed (approved):
Looks good!

mmcallister committed:
adding redirect for Lint appeasement

mmcallister committed:
fixing broken heading link
